src/main/java/com/google/sps/servlets/ShareMeme.java

```diff
@@ -0,0 +1,57 @@
+package com.google.sps.servlets;
+
+import com.google.api.gax.paging.Page;
+
+import com.google.cloud.storage.Blob;
+import com.google.cloud.storage.BlobId;
+import com.google.cloud.storage.BlobInfo;
+import com.google.cloud.storage.Bucket;
+import com.google.cloud.storage.Storage;
+import com.google.cloud.storage.StorageOptions;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.PrintWriter;
+import javax.servlet.ServletException;
+import javax.servlet.annotation.MultipartConfig;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import javax.servlet.http.Part;
+import java.util.*;
+
+@WebServlet("/share")
+@MultipartConfig
+public class ShareMeme extends HttpServlet {
+  public static final String PROJECT_ID = "spring21-sps-43";
+  public static final String BUCKET_NAME = "spring21-sps-43.appspot.com";
+  public static final String LINK = "https://spring21-sps-43.appspot.com/share?meme=";
```

Comment:
To clarify on hard-coded vs relative links, this one is good as it is, because if a user shares the link, we want it to point only towards our live website.